### ISSUE TITLE
Sticky header (and footer)

### ISSUE THREAD

**Issue by katierik** [patternfly/patternfly-next#1526]
We would like an option for a sticky header. This sticky header would remain, well, sticky, while the user scrolls. 
This would likely be useful on mobile as well as desktop.

cc @KendraMar 
cc @ajacobs21e - wondering if you would also need this in designs

**Comment by rachael-phillips** [comment]
Hi @katierik @KendraMar @ajacobs21e 
* What product/project does this impact?
* List any product/project release(s) and timelines your trying to align with
* Are you interested in HTML/CSS only or leveraging a PF-React implementation?

**Comment by ajacobs21e** [comment]
@katierik I included a sticky header in my Solution Engine (Customer Portal) designs but that feature hasn't been reviewed by the team yet as something we want to pursue. 

Another project on the Portal, the Red Hat Dashboard, does include a sticky nav. For your use case, does the header include links / navigation?

**Comment by KendraMar** [comment]
@ajacobs21e @rachael-phillips @katierik 

- RH Cloud Services (aka the old Insights) and all apps that join it (there are at least 6 of them so far)
- September Ansible Fest 
- not sure re third question...I think it would be the Pfly4 team who would recommend the right approach?

@ajacobs21e We'll get back to you re exactly what the customers (John and Logan in the paragraph below) in the pilot test of the RH Cloud Services app were asking to be sticky. Seems like they want only the table rows/data to scroll on the page, so they want to see everything from table headers above frozen on the page (sound right to you, @katierik ?). Allie, have you guys heard this request before from other customers and product groups?  

This is the hifi wireframe version of the working software they're using: https://redhat.invisionapp.com/d/main#/console/15805752/345127480/preview

Here are their comments:

"One complaint is I would like to see the Systems / CVEs buttons at the top of the page be frozen so that when I scroll I can always see those options.  Today I have to scroll back up to the top before I can select them.  This may just be a personal preference, but I see it as a time saver.
John: Do Find a CVE and CVSS base score need to also be frozen?
Logan: CVEs and Systems are critical.  Find a CVE and CVSS base score would be nice, but not a big deal"

**Comment by priley86** [comment]
+1 for adding this feature. I believe OpenShift and friends could use it as well. 

cc: @aljesusg

**Comment by aljesusg** [comment]
+1 for [Kiali Project](https://github.com/kiali/kiali)

**Comment by ajacobs21e** [comment]
@KendraMar I have not heard that feedback before. @rblackbu have you heard anything similar for RHSM? To have the filters or table heading sticky as you scroll through a table?
I think it makes a lot of sense though. My only concern would be the extra space between the CVE and Systems tabs and the top of the table. Have you talked to anyone on the visual design team about options to removing the white space (or in this case grey space) in a sticky situation?

**Comment by rachael-phillips** [comment]
PF4 team will investigate whether sticky header can be complete as a non-breaking change

**Comment by maryclarke** [comment]
Looking for clarification on what header is being requested to be sticky...the masthead or the toolbar above a data table or data list?

**Comment by KendraMar** [comment]
Hi, can we postpone discussion of this one to post summit? We need to investigate it more ourselves.

**Comment by priley86** [comment]
i believe on our side this was in regard to the masthead - can you confirm @aljesusg ? I have checked OpenShift and I do not currently see any `postion: sticky` CSS. It seems the current resolution there was just to allow the page/sidebar contents to scroll independently. So `pf-c-page__main` is set to `overflow-y: auto` for example.

**Comment by serenamarie125** [comment]
We should probably create multiple issues to track individual requests.  For OpenShift & Kiali, having a sticky masthead is desired

**Comment by serenamarie125** [comment]
@katierik can you clarify that for this issue, when you say sticky header, you are talking about the masthead (or something else)?  Thanks!

**Comment by KendraMar** [comment]
@serenamarie125 @katierik See my description of the issue that generated this github issue. Also, see my comment about us postponing or examining the issue further before we do anything.

**Comment by aljesusg** [comment]
yes @priley86 you can test this in [storybook](http://patternfly-react.surge.sh/patternfly-4/demos/pagelayout/examples/page-layout-default-nav)  with the page-layout-default

Here is the gif

![header](https://user-images.githubusercontent.com/3019213/54450371-85d81f00-4750-11e9-97e5-57dbca183437.gif)

**Comment by katierik** [comment]
Following up on this with another ask for "sticky headers", this one for the Drift Analysis application on cloud.redhat.com.
In this case, the table headers are intended to be sticky and stay in view as the user scolls down.
![Xnip2019-12-11_14-32-49](https://user-images.githubusercontent.com/23172038/70653793-2c3f4b00-1c23-11ea-89f5-f14d395aaff3.jpg)

**Comment by mcarrano** [comment]
Transferring this to patternfly-design and assigning to myself. Requires more analysis on what elements need to be sticky and when components that would need to be enhanced.
